src/utils.rs

```diff
@@ -56,6 +56,18 @@ pub fn max_le(a: [u8; 32], b: [u8; 32]) -> [u8; 32] {
 	a
 }
 
+/// Convert nbits to target
+#[allow(dead_code)]
+#[inline]
+pub fn nbits_to_target(nbits: u32) -> [u8; 32] {
+	let mut res = [0; 32];
+	let zero_bytes = 32 - (nbits >> 24) & 0xff;
+	res[(31 - zero_bytes) as usize] = ((nbits >> 16) & 0xff) as u8;
+    res[(31 - zero_bytes - 1) as usize] = ((nbits >> 8) & 0xff) as u8;
+	res[(31 - zero_bytes - 2) as usize] = (nbits & 0xff) as u8;
```

Comment:
This isn't safe, you need to handle the case where the shift is further right see https://github.com/bitcoin/bitcoin/blob/v0.16.2/src/arith_uint256.cpp#L210
Also, maybe handle the negative case?

Reply:
Yeah, i will fix this.